adds ability to declaratively fail on null field

da3mon opened the pull request:
@bryanduxbury

Commit:
adds ability to declaratively fail on null field

bryanduxbury commented on src/main/java/com/squareup/cascading_helpers/filter/FailOnNull.java:
```diff
@@ -0,0 +1,27 @@
+
+  @Override public boolean isRemove(FlowProcess flowProcess, FilterCall filterCall) {
+    if (filterCall.getArguments().getObject(0) == null) {
+      throw new RuntimeException(errorText);
```
Would it be more appropriate to throw a NullPointerException? You're adding user message text already, so it's no more oblique.

bryanduxbury commented on src/main/java/com/squareup/cascading_helpers/pump/Pump.java:
```diff
@@ -102,6 +103,10 @@ public Pump each(Filter filter, String... args) {
     return new FilterPump(this, filter, args);
   }
 
+  public Pump failOnNull(String errorText, String... args) {
```
The underlying FailOnNull only interacts with one tuple field, so you should change this signature from String... args to String field

bryanduxbury commented on src/test/java/com/squareup/cascading_helpers/pump/TestPump.java:
```diff
@@ -181,6 +194,23 @@ public void testEachFunction() throws IOException {
+
+      tap = getSequenceFileTap(NULL_INPUT_PATH);
+      CascadingHelper.get().getFlowConnector().connect(tap, getOutTap(), p).complete();
+      fail();
```
Add a message to this fail like "Expected an exception, but none was thrown"

bryanduxbury commented:
Remember to update the version number.

Commit:
addresses lack of failure message

Commit:
adds support for specifying fields

provide ability to guarantee that the fields named in args are not null
before proceeding

Commit:
improves documentation

Commit:
inlines local variable from field

Commit:
stops unneeded warnings

bryanduxbury commented:
LGTM except still need to bump version number.

Commit:
bumps version
